index.js

```diff
@@ -10,40 +7,36 @@ var WindowsBalloon = require('./notifiers/balloon');
 
 var options = { withFallback: true };
 
-var osType = utils.isWSL() ? 'WSL' : os.type();
-
-switch (osType) {
-  case 'Linux':
-    module.exports = new NotifySend(options);
-    module.exports.Notification = NotifySend;
-    break;
-  case 'Darwin':
-    module.exports = new NotificationCenter(options);
-    module.exports.Notification = NotificationCenter;
-    break;
-  case 'Windows_NT':
-    if (utils.isLessThanWin8()) {
-      module.exports = new WindowsBalloon(options);
-      module.exports.Notification = WindowsBalloon;
-    } else {
-      module.exports = new WindowsToaster(options);
-      module.exports.Notification = WindowsToaster;
-    }
-    break;
-  case 'WSL':
-    module.exports = new WindowsToaster(options);
-    module.exports.Notification = WindowsToaster;
-    break;
-  default:
-    if (os.type().match(/BSD$/)) {
-      module.exports = new NotifySend(options);
-      module.exports.Notification = NotifySend;
-    } else {
-      module.exports = new Growl(options);
-      module.exports.Notification = Growl;
-    }
+function getOsType () {
+  var utils = require('./lib/utils');
+  var osType = require('os').type();
+
+  if (osType.match(/BSD$/)) {
+    osType === 'BSD';
+  } else if (osType === 'Windows_NT' && utils.isLessThanWin8()) {
+    osType === 'Windows_7_and_Below';
+  } else if (utils.isWSL()) {
+    osType = 'WSL';
+  }
+
+  return osType.toLowerCase();
```

Comment:
Store all of the logic for determining the OS in one function. Keeps all the related code together. Also easier for unit testing if you ever decide to do that.